test/minikube/minikube_domain_resync_test.go

```diff
@@ -101,6 +101,55 @@ func checkArchives(t *testing.T, namespaceName string, adminPod string, numExpec
 	return
 }
 
+func TestAdminScaleDown(t *testing.T) {
+	testlib.AwaitTillerUp(t)
+	defer testlib.VerifyTeardown(t)
+	defer testlib.Teardown(testlib.TEARDOWN_ADMIN)
+
+	helmChartReleaseName, namespaceName := testlib.StartAdmin(t, &helm.Options{
+		SetValues: map[string]string{
+			"admin.replicas": "2",
+		},
+	}, 2, "")
+
+	adminStatefulSet := helmChartReleaseName + "-nuodb-cluster0"
+	admin0 := adminStatefulSet + "-0"
+	admin1 := adminStatefulSet + "-1"
+
+	// scale down Admin StatefulSet
+	options := k8s.NewKubectlOptions("", "")
+	options.Namespace = namespaceName
+	k8s.RunKubectl(t, options, "scale", "statefulset", adminStatefulSet, "--replicas=1")
+
+	// wait for scaled-down Admin to show as "Disconnected"
+	testlib.Await(t, func() bool {
+		output, _ := k8s.RunKubectlAndGetOutputE(t, options, "exec", admin0, "--",
+			"nuocmd", "show", "domain", "--server-format", "{id} {connected_state}")
+		return strings.Contains(output, admin1+" Disconnected")
+	}, 300*time.Second)
+
+	// wait for scaled-down Admin Pod to be deleted
+	testlib.AwaitNoPods(t, namespaceName, admin1)
+
+	// commit a Raft command to confirm that remaining Admin has consensus
+	k8s.RunKubectl(t, options, "exec", admin0, "--",
+		"nuocmd", "set", "value", "--key", "testKey", "--value", "testValue", "--unconditional")
+
+	// admin1 is still in membership, though it is excluded from consensus;
+	// delete PVC to cause it to be completely removed from the membership;
+	// this should allow the Admin health-check to succeed
```

Comment:
I do not fully understand. My questions are:
1) should there not be an await or something to make sure that the k8s aware action was taken?
2) the comment says that health-check succeeds. Why would it fail before? If it fails, how is it possible to commit a command on line 135?

Reply:
Take a look at the corresponding Fisheye/Crucible to see how the consensus override works. The consensus override is not triggered periodically or by event notifications. The Admin process that is trying to become Raft leader or replicate a command (if it already considers itself leader) will check the stateful K8s client for the conditions I mentioned above, every time that it is evaluating that there is a quorum.